src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2356,6 +2371,30 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
                     this.Release(ref resetOnFailurePoint);
                 }
             }
+
+            static bool isAcceptableNonDeclarationStatement(StatementSyntax statement, bool isScript)
+            {
+                switch (statement?.Kind)
+                {
+                    case null:
+                    case SyntaxKind.LocalDeclarationStatement:
+                    case SyntaxKind.LocalFunctionStatement:
+                    case SyntaxKind.ExpressionStatement when
+                                !isScript &&
+                                // Do not parse a single identifier as an expression statement in a Simple Program, this could be a beginning of a keyword and
+                                // we want completion to offer it.
+                                ((ExpressionStatementSyntax)statement) is var exprStatement &&
```

Comment:
It seems like a null 'statement' is matched here and then we go on to unsafely access members on it.

Reply:
> It seems like a null 'statement' is matched here and then we go on to unsafely access members on it.

See ```case null:``` above.

---
In reply to: [388021510](https://github.com/dotnet/roslyn/pull/42147#discussion_r388021510) [](ancestors = 388021510)